icu4c/source/i18n/smpdtfmt.cpp

```diff
@@ -721,19 +721,29 @@ void SimpleDateFormat::construct(EStyle timeStyle,
 
     UnicodeString timePattern;
     if (timeStyle >= kFull && timeStyle <= kShort) {
+        bool hasRgOrHcSubtag = false;
+        // also use DTPG if the locale has the "rg" or "hc" ("hours") subtag-- even if the overriding region
+        // or hour cycle is the same as the one we get by default, we go through the DateTimePatternGenerator
+        char dummy;
```

Comment:
you do not need char dummy;
Just passing in nullptr instead of &dummy when you call  locale.getKeywordValue

Reply:
Maybe currently, but that behavior is not in the API docs.

Reply:
I'm going to go with Frank here.  We have too many examples of using NULL in this situation already to change it so that it doesn't work (and these APIs are already inconvenient enough).